src/main/sensors/battery.c

```diff
@@ -294,10 +296,10 @@ void updateCurrentMeter(int32_t lastUpdateAt, rxConfig_t *rxConfig, uint16_t dea
         case CURRENT_SENSOR_ESC:
             #ifdef USE_ESC_SENSOR
             if (feature(FEATURE_ESC_SENSOR)) {
-                escSensorData_t escData = getEscSensorData(ESC_SENSOR_COMBINED);
-                if (!escData.stale) {
-                    amperageLatest = escData.current;
-                    mAhDrawn = escData.consumption;
+                escSensorData_t *escData = getEscSensorData(ESC_SENSOR_COMBINED);
+                if (escData->dataAge <= MAX_ESC_BATTERY_AGE) {
```

Comment:
It would make much more sense to be
if (escData->currentRetryCount<= MAX_ESC_BATTERY_RETRY_COUNT) {